### DISCUSSION TITLE
Ultimate set up guide (At least worked for me)

### DISCUSSION THREAD

**Discussion by Bullsito** [OwlGamingCommunity/V#20] [Ideas]
First of all, I don't speak english so if there is any grammar mistake or whatever, don't mind me to correct it.

I spent with my team near a month trying to set up the gamemode and when we were near give up Chaos told me what should we do to make it work and he let me post and reupload the gamemode.

Easy way:

- Just download the gamemode already setted up and run it (Untested, it should be no problem but I can't asure you the 100% working, just remember change your database credentials, Step 3 in Long Way)

 (https://mega.nz/file/EwFRUIoR#ICffRubjlQNyIrXNFmfJvwDi46PI3SrB1HnNwwSfw9Q)

________

Long way:

1. Open GIT Bash -> Move to you prefered directory -> Install GIT LFS (`git lfs install in you bash terminal`) -> Clone this repo with LFS (`git lfs clone https://github.com/OwlGamingCommunity/V.git`) -> It should be created a folder called 'V' -> You can close GIT Bash if you want

2.  Go in to 'V' Folder -> Execute OwlV.sln -> Change the solution config from the toolbar to set it up in DEBUG-X64-BOOT/COOKER/EventComposer

![image](https://user-images.githubusercontent.com/77119291/133907124-f8d6942d-734c-4f28-8d9f-42be115f76e8.png)

3. Open owl.mysql -> Open MySQL.cs and ThreadedMySQL.cs -> Search for `SettingHelpers.GetDevEnvironmentSetting` in both files -> Change the connection parameters if you have others (In my case, my password is 'admin'). **IMPORTANT: Change auth port to 3306 (If you have that port for MySQL), NOT 3307**

![image](https://user-images.githubusercontent.com/77119291/133909099-aa00cc45-d05c-465f-a086-114faf172462.png)

![image](https://user-images.githubusercontent.com/77119291/133909113-416e59e0-c106-43a4-a86a-52c5c893af35.png)

4.  Open EventComposer -> Open EventComposer.cs -> Go to line 90 (or wherever 'UI Descriptors' starts) -> Change ".." to "..\\..\\..\\.." (Somehow, the original path is incorrect and with that change we can put the correct one) -> Do the same with Network Descriptors (Line 120) -> Search for "client & server" or "owl_core" or "owl_core.client" (Line 365) and change "owl_core" or "owl_core.client" to "..\..\..\owl_core" or "..\..\..\owl_core.client" (Yeah, the path is fuck up too there)

![image](https://user-images.githubusercontent.com/77119291/133909344-9b75d335-2b9b-4bdf-948a-9138cc9ad8f9.png)

![image](https://user-images.githubusercontent.com/77119291/133909353-4d1d9aaf-21fd-409b-82b7-50da0a1fc0a5.png)

5. Rebuild ALL the solution -> You should have now a folder called 'Output' -> Paste all the content from 'RAGE-MP\Server-files' to 'Output\Debug' -> Go to Output\Debug\Plugins and delete bridge.dll

![image](https://user-images.githubusercontent.com/77119291/133909396-1401ecb2-b134-4116-a6ae-25f9ae871ef0.png)

![image](https://user-images.githubusercontent.com/77119291/133909407-be4a102c-1d92-4f96-aa2b-7dc0b426568f.png)

![image](https://user-images.githubusercontent.com/77119291/133909587-555d2674-4974-4d2c-8404-d3a1aae8481d.png)

![image](https://user-images.githubusercontent.com/77119291/133909619-af5e4c1a-c7bd-4d49-a869-a583f849eedd.png)

6.  Run EventComposer from Visual Studio with Debug-x64-EventComposer

![image](https://user-images.githubusercontent.com/77119291/133909659-71dc5738-ea4e-4076-9e54-3d7a1838ed4a.png)

![image](https://user-images.githubusercontent.com/77119291/133909692-86a69dd6-0165-4d42-9d1b-118ce003fca3.png)

7. Run Cooker from Visual Studio with Debug-x64-Cooker (you shouldn't have any errors with this)

![image](https://user-images.githubusercontent.com/77119291/133909738-79453b8d-06f1-49ef-af3d-01029f211c1b.png)

8. Make sure your database is online (I use PHPMyAdmin)

![image](https://user-images.githubusercontent.com/77119291/133910071-5afca6c1-c3f7-4cd9-9fc0-95aa19bad862.png)

9. Run Boot from Visual Studio with Debug-x64-Boot **PAY ATENTTION IF YOU MAKE IT TO THIS STEP**

![image](https://user-images.githubusercontent.com/77119291/133909766-b44b2323-9c3d-4b2b-bdb3-2aba4ab14c08.png)

10. You may think it's done, the server is up and running but if you try to enter, it's an empty gamemode, you can do nothing, you have to enable C#

![image](https://user-images.githubusercontent.com/77119291/133909944-c93c3549-c68c-4b77-a979-fbca69678ec1.png)

11. Go to Ouput\Debug -> Open conf.json with your text editor -> Change `"csharp" : "disabled"` to `"csharp" : "enabled"` and save

![image](https://user-images.githubusercontent.com/77119291/133909979-b0b159f7-6132-4230-9288-7492225d4e86.png)

![image](https://user-images.githubusercontent.com/77119291/133910013-f651371d-64f5-4f0d-b3b3-4c363940b90d.png)

12. Run again Boot -> The good news it's Script_loader is working but the but news it's you now have an error "Assembly with the same name is already loaded" -> Go to Output\Debug\dotnet\resources\deps_rageoverrides and delete "Newtonsoft.json.dll" (its a shortcut)

![image](https://user-images.githubusercontent.com/77119291/133910062-ad16bfbb-38bf-47e9-b862-9267ac5115be.png)

![image](https://user-images.githubusercontent.com/77119291/133910057-ecbe601e-894a-4f84-97a8-21149258397d.png)

13. Before running Boot again, you should delete "cooked_maps" shortcut from Output\Debug -> Paste the whole folder from the first link where is the whole gamemode or in this link that its only the folder (https://mega.nz/file/Y091nK4Z#UfCKKCQO08SraGbBEPY1TcCqmrGrCYAmgbcJYyn834M). 

The first 'folder' (shortcut) doesn't let you get in, so the server can't do it either so you force the access with the entire folder with its content.

![image](https://user-images.githubusercontent.com/77119291/133910114-17ec7fec-3f4b-458d-96b1-33a28869d4a3.png)

![image](https://user-images.githubusercontent.com/77119291/133910129-fc2baa0f-67a1-4977-a062-b9386bcce160.png)

(You should generate too this folder without download anything with Release-x64-Cooker but I didn't test it well while I set up this gamemode, so better safe than sorry)

14. Run AGAIN and for the last time Debug-x64-Boot -> Congrats, you server should be running

![image](https://user-images.githubusercontent.com/77119291/133910184-46484717-1cf7-4239-8049-d9abfbd9ba85.png)

15. Try to connect to the server

![image](https://user-images.githubusercontent.com/77119291/133910258-b5236fbd-e23e-4a51-b5d8-992961bbd5b9.png)

_____

If you need help, try to reach Owl Discord or here in their Github, I'm just a regular guy who could start this gamemode after a month trying and trying with the help of some of this original devs.

I ONLY COULD SET IT AND START IT, I DIDN'T PLAY OR TEST THIS GAME MODE AT THE TIME OF WRITING THIS GUIDE

**Comment by ismatuner** [comment]
Many thanks but I got this error:

Table 'gtav.custom_interior_maps' doesn't exist
Table 'gtav.custom_interior_maps' doesn't exist
Table 'gtav.custom_interior_maps' doesn't exist
Table 'gtav.custom_interior_maps' doesn't exist
Table 'gtav.custom_interior_maps' doesn't exist
[ERROR] MySQL Query Error [GAME]: Table 'gtav.custom_interior_maps' doesn't exist
CALL STACK:    at System.Environment.get_StackTrace()
   at PrintLogger.LogMessage(ELogSeverity logSeverity, String strFormat, Object[] strParams)
   at Database.MySQLInstance.QueryGame(String strQuery, Object[] formatParams)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Database.MySQLInstance.QueryGame(String strQuery, Object[] formatParams)
   at Database.LegacyFunctions.GetAllCustomInteriors()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Database.LegacyFunctions.GetAllCustomInteriors()
   at MapLoader.Init()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at MapLoader.Init()
   at MapLoader..ctor()
   at Core..ctor()
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
   at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type)
   at OwlScriptManager..ctor()
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
   at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at GTANetworkInternals.GameServer.InstantiateScripts(Assembly targetAssembly, String resourceName)+MoveNext()
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at GTANetworkInternals.GameServer.StartResource(String resourceName, String parent)
   at GTANetworkInternals.GameServer.Start(IEnumerable`1 filterscripts)
   at GTANetworkInternals.GameServer.InitializeBootstrapper()
   at Bootstrapper.EventInternalClass.InitializeAPI(UInt32 sectionSize, UInt64* names, IntPtr method)
Press any key to continue...

**Reply by Bullsito** [reply]
Did you check you got the right connection parameters to you database? Is the GTAV database properly imported? Other fella told me he had a similar error and I don't know why, it works for me, it seems to be a local database error.

**Reply by KevinGarciaR** [reply]
@ismatuner The table custom_interior_maps doesn't exists in the database... I had similar problems like that, so I had to install like 5-6 tables manually.  You must guide yourself with the logs.

**Reply by Yannick-DG** [reply]
```
CREATE TABLE `custom_interior_maps` (
  `id` int(10) UNSIGNED NOT NULL,
  `property_id` int(10) NOT NULL,
  `marker_x` float NOT NULL,
  `marker_y` float NOT NULL,
  `marker_z` float NOT NULL,
  `uploaded_at` datetime DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

ALTER TABLE `custom_interior_maps`
  ADD PRIMARY KEY (`id`),
  ADD KEY `custom_interior_maps_property_id_created_at_index` (`property_id`,`uploaded_at`);

ALTER TABLE `custom_interior_maps`
  ADD CONSTRAINT `custom_interior_maps_property_id_foreign` FOREIGN KEY (`property_id`) REFERENCES `properties` (`id`) ON DELETE CASCADE;
```

**Reply by heartsiddharth** [reply]
can you help with the .msql database structre file to import if you can purge the data that will be really helpful i just need to import the structre.

**Comment by heartsiddharth** [comment]
Could someone please help me with the database structure file , i just need the structure file as i am getting everytime the errors of tables does not exist please help 


MySQL Query Error [GAME]: Table 'gtav.world_blips' doesn't exist

@yannickboy15 

@Bullsito 

i am trying to easy version one and it is giving me erros if a .mysql can be geneerated it will solve lot of problems.  please help

**Reply by Bullsito** [reply]
Hi, sorry for the late reply.

Did you import correctly the .sql files into your database? Did you set your own connection parameters?

**Reply by Paul0405** [reply]
@Bullsito

where can i find the sql?

**Reply by Bullsito** [reply]
> @Bullsito
> 
> where can i find the sql?

Data folder, there are two .sql files

**Comment by woozievert** [comment]
could not find Mysql.Data.resources.dll

**Reply by Aurydevelopment** [reply]
me too

**Comment by Aurydevelopment** [comment]
My database has no password. I leave this blank and I get this error. What is the reason? 

![Screenshot_4](https://user-images.githubusercontent.com/64798519/141836461-69de9bff-2bae-4252-b844-10d534ca7ff5.png)

![Screenshot_3](https://user-images.githubusercontent.com/64798519/141836453-949897de-1454-453a-8873-b0bc8d511740.png)

**Comment by Bullsito** [comment]
The database credential thing stills being a mysterie, no fix at the moment for that.

**Comment by Yannick-DG** [comment]
Most of you guys are probably running XAMPP or shit. Check this out and reset your MySQL root password to like root or something you guys know.

https://stackoverflow.com/questions/24566453/resetting-mysql-root-password-with-xampp-on-localhost

After that, hop into your environment variables on your computer or use the built in solution environment variables of Visual Studio and enter all the credentials in there. Also make sure that port 3307 of the AUTH database is set to 3306 if you host it all on 1 server.

**Reply by Aurydevelopment** [reply]
I'm using XAMPP. What program would you recommend I use to avoid this mysql problem? I can't connect to mysql.

**Reply by Yannick-DG** [reply]
XAMPP is fine, just make sure you've set it up right otherwise the start is already messed up.

Alternative would be a regular MySQL Server or whatever you'd like, but for development purposes XAMPP does the trick.

**Reply by heartsiddharth** [reply]
Can anyone please help me with .mysql file  which has all the structures ?

**Reply by Aurydevelopment** [reply]
> XAMPP is fine, just make sure you've set it up right otherwise the start is already messed up.
> 
> Alternative would be a regular MySQL Server or whatever you'd like, but for development purposes XAMPP does the trick.

Other rpg gamemodes are working. But in this game mode, the connection to the database doesn't work. I added in mysql password but it doesn't work.

**Comment by miqadze** [comment]
`Could not load type 'GTANetworkAPI.Vector3' from assembly 'owl_servershared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' due to value type mismatch.`

any ideas what the hell is going on?

**Comment by mircea1223** [comment]
try to use his credential on your database. user: root pass: admin

**Reply by Bullsito** [reply]
The (database) credentials thing in this gamemode its a little tricky.

If you don't have the environment variables declare, you had to use the default credentials, which I think its admin:example or root:example or something like that, I don't recall if they work with root:admin (therefore, you have to configure your database to that credentials).

If you wanna to use YOUR OWN credentials, you have to set it as environment variables, at Windows its easy, I never did that at Linux but its Linux so there should be a lot of guides on the internet.

**Comment by SirNeoTheMatrix** [comment]
All is good so far on locale ... but thing is i dont know what causing it in game i have 2 big problems so far... that not letting me continue
1) when rain is starting - ALL screen is in fog too much. u cant even see 1 cm from head... 
2) makeint is giving error because of 3 argument, bool because it ask something about blips and there is no info of what blip name is..

**Reply by Bullsito** [reply]
1- There was a few weathers that are so weird, I guess it tas 9,10,11. Anyways, you can test them with /setw
2- I guess, as the syntax message might show, to make a global blip for the property or something like that, as far as I remember, some bool parameters must be passed as 'true' or 'false' (doesn't work with 1 or 0)

**Comment by ins72** [comment]
can i contact you on discord or something?

**Comment by Paul0405** [comment]
Could anyone show me where the SQL of the database is?

**Reply by Bullsito** [reply]
Data folder, there are two .sql files

**Comment by PolishDywan** [comment]
![image](https://user-images.githubusercontent.com/65501143/169976446-d9baa7ea-8dba-4018-83bd-24c745edb151.png)

**Comment by AxelKlu** [comment]
How to i start the server ./V/OwlV.sin does not work

**Comment by AxelKlu** [comment]
Can someone help me with the installation, i pay, contact me on discord freestyleaxel#6743

@Bullsito

**Comment by AxelKlu** [comment]
userputtyftpvgls98wx@inf04-scyte-379:~$ ./V/OwlV.sln
./V/OwlV.sln: line 1: $'\357\273\277\r': command not found
./V/OwlV.sln: line 2: Microsoft: command not found
./V/OwlV.sln: line 4: VisualStudioVersion: command not found
./V/OwlV.sln: line 5: MinimumVisualStudioVersion: command not found
./V/OwlV.sln: line 6: syntax error near unexpected token `"{9A19103F-16F7-4668-BE54-9A1E7A4F7556}"'
./V/OwlV.sln: line 6: `Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "owl_mysql", "So'rce\owl_mysql\owl_mysql.csproj", "{6531FF70-1D34-456B-A9F3-566F6BDA9702}"


what do i do about this error?

**Comment by V1ruSsSsS** [comment]
How to set it up on linux (Ubuntu 22.04 LT)? Is it even possible or should i change the contract from linux server to windows?

I've done multiple tries to set it up and the best i could get was when i started the server and saw the chat that's in Owl scripts. Nothing else. Nothing else loaded, no server-side scripts were loaded. Please help. Reach me out here on by discord - v1russss
